### ISSUE TITLE
UnhandledPromiseRejectionWarning: DBusError: Does Not Exist

### ISSUE THREAD

**Issue by johro897** [icanos/hassio-plejd#104]
After a new install of Hassio, 0.114.4 I got the following issue and plejd doesnt work. Prior this, with another installation, it worked fine. only component changed is the SD Card and the Hassio version from 0.112.1

(node:293) UnhandledPromiseRejectionWarning: DBusError: Does Not Exist
at methodReturnHandlers.(anonymous function) (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
at EventEmitter.MessageBus.conn.on (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
at EventEmitter.emit (events.js:198:13)
at /plejd/node_modules/dbus-next/lib/connection.js:112:14
at Socket. (/plejd/node_modules/dbus-next/lib/message.js:63:9)
at Socket.emit (events.js:198:13)
at emitReadable (_stream_readable.js:555:12)
at process._tickCallback (internal/process/next_tick.js:63:19)
(node:293) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:293) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by icanos** [comment]
Seems like DBus isn't recognized? Do other BLE stuff work?

**Comment by johro897** [comment]
Not using it for any other BLE

**Comment by johro897** [comment]
tried this and it seems like BLE works:
bluetoothctl
Agent registered
[CHG] Controller B8:27:EB:2F:F0:E0 Pairable: yes
[CHG] Controller 00:1A:7D:DA:71:08 Pairable: yes
[bluetooth]# scan on
Discovery started
[CHG] Controller 00:1A:7D:DA:71:08 Discovering: yes
[NEW] Device 5C:F9:38:BE:96:0D 5C-F9-38-BE-96-0D
[NEW] Device D1:4A:4F:34:26:97

**Comment by icanos** [comment]
Make sure that dbus is installed

**Comment by martach** [comment]
Has this issue been resolved? After an update I get a similar error:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Hass.io Add-on: Plejd
 Adds support for the Swedish home automation devices from Plejd.
-----------------------------------------------------------
 Add-on version: 0.4.7
 There is an update available for this add-on!
 Latest add-on version: null
 Please consider upgrading as soon as possible.
parse error: Expected string key before ':' at line 1, column 4
[19:32:57] ERROR: Unknown HTTP error occured
 System:   (armv7 / raspberrypi4)
 Home Assistant version: 0.118.0
 Supervisor version: 2020.11.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[19:32:58] INFO: Starting the Plejd service...
[19:32:58] INFO: Verified permissions on startup script
[19:32:58] INFO: Executing startup script
[19:32:59] INFO: Wrote plejd.json
[19:32:59] INFO: Running add-on
starting Plejd add-on v. 0.4.7
plejd-api: login()
plejd-api: logging into XXXXXXXXXXXXXXXXX
plejd-api: got session token response
plejd-api: getSites()
plejd-api: got site list response
plejd-api: getSite(...)
plejd-api: got site details response
wireEvents()
init()
plejd-ble: disconnecting /org/bluez/hci0/dev_D4_DC_DD_EE_D6_4E
plejd-mqtt: connected to mqtt.
plejd-mqtt: discovered light (DIM-01) named Köksö with PID 21.
plejd-mqtt: discovered light (DIM-01) named Köks-L with PID 11.
plejd-mqtt: discovered light (DIM-01) named Spottar nedre badrum with PID 26.
plejd-mqtt: discovered light (DIM-01) named Spottar övre badrum with PID 16.
plejd-mqtt: discovered light (DIM-01) named Köksbord with PID 30.
plejd-mqtt: discovered light (DIM-01) named hörn kök/lekrum with PID 31.
plejd-mqtt: discovered light (DIM-01) named Kamin with PID 33.
plejd-mqtt: discovered light (DIM-01) named Johans rum with PID 36.
plejd-mqtt: discovered light (DIM-01) named Teve with PID 34.
plejd-mqtt: discovered switch (Scene) named Ljust with PID 1.
plejd-mqtt: discovered switch (Scene) named Mörkt with PID 3.
plejd-mqtt: discovered switch (Scene) named Läsa soffa with PID 5.
plejd-mqtt: discovered switch (Scene) named Middag with PID 2.
plejd-mqtt: discovered switch (Scene) named Mys vardagsrum with PID 6.
plejd-ble: connecting to /org/bluez/hci0/dev_D4_DC_DD_EE_D6_4E
onDeviceConnected()
trying 10 characteristics
authenticate()
startPing()
startWriteQueue()
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 6 byte(s) of data to Plejd
error: plejd ping failed
plejd-ble: ping failed, reconnecting.
init()
error: writing to plejd: DBusError: Not connected
plejd-ble: ping failed, reconnecting.
init()
(node:291) UnhandledPromiseRejectionWarning: DBusError: Does Not Exist
    at _methodReturnHandlers.(anonymous function) (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
    at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
    at EventEmitter.MessageBus.conn.on (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
    at EventEmitter.emit (events.js:198:13)
    at /plejd/node_modules/dbus-next/lib/connection.js:112:14
    at Socket.<anonymous> (/plejd/node_modules/dbus-next/lib/message.js:63:9)
    at Socket.emit (events.js:198:13)
    at emitReadable_ (_stream_readable.js:555:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
(node:291) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:291) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
plejd-ble: connecting to /org/bluez/hci0/dev_D4_DC_DD_EE_D6_4E
```

**Comment by martach** [comment]
Is this possibly connected to a system upgrade (I am at a RPi4 with Operating System
HassOS 4.17 and Supervisor 2020.11.0)? I tried to resolve the issue by reinstalling the addon, but when I try to install it, it fails with this error:
```
The command '/bin/bash -o pipefail -c apk add --no-cache --virtual .build-dependencies g++=8.3.0-r0 gcc=8.3.0-r0 libc-dev=0.7.1-r0 linux-headers=4.19.36-r0 make=4.2.1-r2 python bluez eudev-dev=3.2.8-r0 && apk add --no-cache git nodejs npm dbus-dev glib-dev && npm config set unsafe-perm true' returned a non-zero code: 68
```
The supervisor log shows the following information:
```
---> Using cache
 ---> 2997a1f879c3
Step 14/28 : RUN apk add --no-cache jq
 ---> Using cache
 ---> 72bebfa617e9
Step 15/28 : RUN   apk add --no-cache --virtual .build-dependencies   g++=8.3.0-r0   gcc=8.3.0-r0   libc-dev=0.7.1-r0   linux-headers=4.19.36-r0   make=4.2.1-r2   python   bluez   eudev-dev=3.2.8-r0     && apk add --no-cache   git   nodejs   npm   dbus-dev   glib-dev     && npm config set unsafe-perm true
 ---> Running in ac81d8b13e6b
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/main/armv7/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/community/armv7/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/edge/main/armv7/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/edge/community/armv7/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/edge/testing/armv7/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:


  musl-1.1.22-r3:
    breaks: musl-dev-1.1.22-r4[musl=1.1.22-r4]
    satisfies: world[musl=1.1.22-r3]
               libbsd-0.9.1-r0[musl>=1.1.16-r22]
               libbsd-0.9.1-r0[so:libc.musl-armv7.so.1]
               apk-tools-2.10.4-r2[so:libc.musl-armv7.so.1]
               kmod-24-r1[so:libc.musl-armv7.so.1]
               pkgconf-1.6.1-r1[so:libc.musl-armv7.so.1]
               libblkid-2.33.2-r0[so:libc.musl-armv7.so.1]
               gcc-8.3.0-r0[so:libc.musl-armv7.so.1]
               g++-8.3.0-r0[so:libc.musl-armv7.so.1]
               tzdata-2019c-r0[so:libc.musl-armv7.so.1]
               bluez-5.50-r4[so:libc.musl-armv7.so.1]
               polkit-0.116-r0[so:libc.musl-armv7.so.1]
               bash-5.0.0-r0[so:libc.musl-armv7.so.1]
               ca-certificates-20190108-r0[so:libc.musl-armv7.so.1]
               libxau-1.0.9-r0[so:libc.musl-armv7.so.1]
               eudev-3.2.8-r0[so:libc.musl-armv7.so.1]
               busybox-1.30.1-r2[so:libc.musl-armv7.so.1]
               consolekit2-1.2.1-r1[so:libc.musl-armv7.so.1]
               binutils-2.32-r0[so:libc.musl-armv7.so.1]
               sqlite-libs-3.28.0-r3[so:libc.musl-armv7.so.1]
               libuuid-2.33.2-r0[so:libc.musl-armv7.so.1]
               curl-7.66.0-r0[so:libc.musl-armv7.so.1]
               nspr-4.20-r0[so:libc.musl-armv7.so.1]
               mpfr3-3.1.5-r1[so:libc.musl-armv7.so.1]
               eudev-libs-3.2.8-r0[so:libc.musl-armv7.so.1]
               linux-pam-1.3.0-r1[so:libc.musl-armv7.so.1]
               libx11-1.6.12-r0[so:libc.musl-armv7.so.1]
               libxcb-1.13.1-r0[so:libc.musl-armv7.so.1]
               ncurses-libs-6.1_p20190518-r0[so:libc.musl-armv7.so.1]
               expat-2.2.8-r0[so:libc.musl-armv7.so.1]
               musl-utils-1.1.22-r3[so:libc.musl-armv7.so.1]
               nghttp2-libs-1.39.2-r0[so:libc.musl-armv7.so.1]
               gdbm-1.13-r1[so:libc.musl-armv7.so.1]
               python2-2.7.18-r0[so:libc.musl-armv7.so.1]
               libstdc++-8.3.0-r0[so:libc.musl-armv7.so.1]
               dbus-1.12.16-r1[so:libc.musl-armv7.so.1]
               libbz2-1.0.6-r7[so:libc.musl-armv7.so.1]
               isl-0.18-r0[so:libc.musl-armv7.so.1]
               libcurl-7.66.0-r0[so:libc.musl-armv7.so.1]
               mpc1-1.1.0-r0[so:libc.musl-armv7.so.1]
               icu-libs-64.2-r1[so:libc.musl-armv7.so.1]
               pcre-8.43-r0[so:libc.musl-armv7.so.1]
               oniguruma-6.9.3-r0[so:libc.musl-armv7.so.1]
               mozjs60-60.8.0-r0[so:libc.musl-armv7.so.1]
               libffi-3.2.1-r6[so:libc.musl-armv7.so.1]
               zlib-1.2.11-r1[so:libc.musl-armv7.so.1]
               libmount-2.33.2-r0[so:libc.musl-armv7.so.1]
               glib-2.60.4-r0[so:libc.musl-armv7.so.1]
               libgomp-8.3.0-r0[so:libc.musl-armv7.so.1]
               libintl-0.19.8.1-r4[so:libc.musl-armv7.so.1]
               readline-8.0.0-r0[so:libc.musl-armv7.so.1]
               make-4.2.1-r2[so:libc.musl-armv7.so.1]
               libxdmcp-1.1.3-r0[so:libc.musl-armv7.so.1]
               libgcc-8.3.0-r0[so:libc.musl-armv7.so.1]
               
libssl1.1-1.1.1d-r0[so:libc.musl-armv7.so.1]
               alpine-baselayout-3.1.2-r0[so:libc.musl-armv7.so.1]
               libcrypto1.1-1.1.1d-r0[so:libc.musl-armv7.so.1]
               libatomic-8.3.0-r0[so:libc.musl-armv7.so.1]
               xz-libs-5.2.4-r0[so:libc.musl-armv7.so.1]
               dbus-libs-1.12.16-r1[so:libc.musl-armv7.so.1]
               scanelf-1.2.3-r0[so:libc.musl-armv7.so.1]
               gmp-6.1.2-r1[so:libc.musl-armv7.so.1]
               jq-1.6-r0[so:libc.musl-armv7.so.1]
Removing intermediate container ac81d8b13e6b
```

**Comment by martach** [comment]
Both the build error and the Dbus error have been resolved for me after the recent fix. Thank you.

**Comment by JohnPhoto** [comment]
This is the multi-init problem again, see #137

**Comment by SweVictor** [comment]
So this can be closed since #137 is merged?

**Comment by icanos** [comment]
Please reopen the issue if the problems continue. Thanks!
